From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Thu, 28 May 2026 22:02:57 +0000
Subject: [PATCH] (Yukigram) Branding


diff --git a/Telegram/SourceFiles/boxes/about_box.cpp b/Telegram/SourceFiles/boxes/about_box.cpp
index 3a488ae04c..cb3f6da6c3 100644
--- a/Telegram/SourceFiles/boxes/about_box.cpp
+++ b/Telegram/SourceFiles/boxes/about_box.cpp
@@ -12,6 +12,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "core/file_utilities.h"
 #include "core/update_checker.h"
 #include "lang/lang_keys.h"
+#include "yukigram/lang.h"
 #include "ui/boxes/confirm_box.h"
 #include "ui/painter.h"
 #include "ui/rect.h"
@@ -34,10 +35,8 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 namespace {
 
 rpl::producer<TextWithEntities> Text1() {
-	return tr::lng_about_text1(
-		lt_api_link,
-		tr::lng_about_text1_api(tr::url(u"https://core.telegram.org/api"_q)),
-		tr::marked);
+	return rktre("about/1/text",
+		{ "tdesktop_link", Ui::Text::Link(ktr("about/1/tdesktop"), "https://desktop.telegram.org/") });
 }
 
 rpl::producer<TextWithEntities> Text2() {
@@ -54,10 +53,9 @@ rpl::producer<TextWithEntities> Text2() {
 }
 
 rpl::producer<TextWithEntities> Text3() {
-	return tr::lng_about_text3(
-		lt_faq_link,
-		tr::lng_about_text3_faq(tr::url(telegramFaqLink())),
-		tr::marked);
+	return rktre("about/3/text",
+		{ "channel_link", Ui::Text::Link(ktr("about/3/channel"), "https://t.me/yukigram") },
+		{ "faq_link", Ui::Text::Link(tr::lng_about_text3_faq(tr::now), telegramFaqLink()) });
 }
 
 } // namespace
diff --git a/Telegram/SourceFiles/history/history_item_helpers.cpp b/Telegram/SourceFiles/history/history_item_helpers.cpp
index 1af7166acc..3ee4f879f1 100644
--- a/Telegram/SourceFiles/history/history_item_helpers.cpp
+++ b/Telegram/SourceFiles/history/history_item_helpers.cpp
@@ -52,6 +52,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "ui/widgets/checkbox.h"
 #include "ui/item_text_options.h"
 #include "lang/lang_keys.h"
+#include "yukigram/lang.h"
 
 namespace {
 
@@ -1317,10 +1318,7 @@ void CheckPollVoteNotificationSchedule(
 }
 
 [[nodiscard]] TextWithEntities UnsupportedMessageText() {
-	const auto siteLink = u"https://desktop.telegram.org"_q;
-	auto result = TextWithEntities{
-		tr::lng_message_unsupported(tr::now, lt_link, siteLink)
-	};
+	auto result = ktre("unsupported-message/text");
 	TextUtilities::ParseEntities(result, Ui::ItemTextNoMonoOptions().flags);
 	result.entities.push_front(
 		EntityInText(EntityType::Italic, 0, result.text.size()));
@@ -1334,9 +1332,9 @@ HistoryMessageMarkupData UnsupportedMessageMarkup() {
 	auto row = std::vector<Button>();
 	row.emplace_back(
 		Button::Type::Url,
-		tr::lng_update_telegram(tr::now),
+		ktr("unsupported-message/button/text"),
 		Button::Visual(),
-		QByteArray("https://desktop.telegram.org"));
+		QByteArray("https://t.me/yukigram"));
 	markup.rows.push_back(std::move(row));
 	return markup;
 }
diff --git a/Telegram/SourceFiles/intro/intro_start.cpp b/Telegram/SourceFiles/intro/intro_start.cpp
index a871886d76..7f27e438ec 100644
--- a/Telegram/SourceFiles/intro/intro_start.cpp
+++ b/Telegram/SourceFiles/intro/intro_start.cpp
@@ -8,6 +8,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "intro/intro_start.h"
 
 #include "lang/lang_keys.h"
+#include "yukigram/lang.h"
 #include "intro/intro_qr.h"
 #include "intro/intro_phone.h"
 #include "ui/widgets/buttons.h"
@@ -25,7 +26,7 @@ StartWidget::StartWidget(
 : Step(parent, account, data, true) {
 	setMouseTracking(true);
 	setTitleText(rpl::single(u"Yukigram"_q));
-	setDescriptionText(tr::lng_intro_about());
+	setDescriptionText(ktr("intro/about"));
 	show();
 }
 
