From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Sun, 26 Apr 2026 22:07:47 +0300
Subject: [PATCH] (Yukigram) Options framework


diff --git a/Telegram/SourceFiles/core/launcher.cpp b/Telegram/SourceFiles/core/launcher.cpp
index 39ad62968d..1267f6f7b3 100644
--- a/Telegram/SourceFiles/core/launcher.cpp
+++ b/Telegram/SourceFiles/core/launcher.cpp
@@ -19,6 +19,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "base/concurrent_timer.h"
 #include "base/options.h"
 #include "yukigram/appid.h"
+#include "yukigram/options_runtime.h"
 
 #include <QtCore/QLoggingCategory>
 #include <QtCore/QStandardPaths>
@@ -388,6 +389,7 @@ int Launcher::exec() {
 	// Must be started before Platform is started.
 	Logs::start();
 	base::options::init(cWorkingDir() + "tdata/experimental_options.json");
+	Yukigram::Options::init();
 
 	// Must be called after options are inited.
 	initHighDpi();
@@ -629,6 +631,7 @@ void Launcher::processArguments() {
 int Launcher::executeApplication() {
 	FilteredCommandLineArguments arguments(_argc, _argv);
 	Sandbox sandbox(arguments.count(), arguments.values());
+	Yukigram::Options::link(); // call after QApplication init
 	Ui::MainQueueProcessor processor;
 	base::ConcurrentTimerEnvironment environment;
 	return sandbox.start();
diff --git a/Telegram/SourceFiles/settings/settings_experimental.cpp b/Telegram/SourceFiles/settings/settings_experimental.cpp
index daa79c8d2f..fe0c3cabcc 100644
--- a/Telegram/SourceFiles/settings/settings_experimental.cpp
+++ b/Telegram/SourceFiles/settings/settings_experimental.cpp
@@ -6,6 +6,7 @@ For license and copyright information please follow this link:
 https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 */
 #include "settings/settings_experimental.h"
+#include "yukigram/options_runtime.h"
 
 #include "data/components/passkeys.h"
 #include "main/main_session.h"
@@ -183,6 +184,7 @@ void AddOption(
 			return;
 		}
 		option.set(toggled);
+		Yukigram::Options::update_value(option.id().toStdString().c_str(), toggled);
 		if (restarter) {
 			restarter->callOnce(st::settingsButtonNoIcon.toggle.duration);
 		}
diff --git a/Telegram/SourceFiles/yukigram/options.cpp b/Telegram/SourceFiles/yukigram/options.cpp
new file mode 100644
index 0000000000..86b4f76b9f
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/options.cpp
@@ -0,0 +1,118 @@
+#include "yukigram/options.h"
+#include "yukigram/options_database.h"
+#include "yukigram/options_runtime.h"
+#include "base/variant.h"
+#include "yukigram/lang.h"
+
+namespace Yukigram::Options {
+
+std::map<std::string_view, std::variant<Option<bool>, Option<int>, Option<QString>>> &Options() {
+	static std::map<std::string_view, std::variant<Option<bool>, Option<int>, Option<QString>>> m;
+	return m;
+}
+
+std::map<std::string_view, std::vector<std::string_view>> &ByCategory() {
+	static std::map<std::string_view, std::vector<std::string_view>> m;
+	return m;
+}
+
+template<typename T> rpl::variable<T> *make_impl(const char *key, Description<T> &&d) {
+	const auto rv = std::make_shared<rpl::variable<T>>(d.defaultValue);
+
+	const auto base = d.mirrorExperimental ? std::nullopt : std::optional(
+		std::make_shared<
+			base::options::option<T>,
+			base::options::descriptor<T>
+		>({
+			.id = key,
+			.defaultValue = d.defaultValue,
+			.restartRequired = d.restartRequired,
+		})
+	);
+	rpl::lifetime lifetime;
+
+	const auto key_v = std::string_view(key);
+	ByCategory()[d.category].push_back(key_v);
+	Options()[key_v] = Option<T>{
+		.key = key_v,
+		.d = std::move(d),
+		.v = rv,
+		._base = base,
+		._lifetime = std::move(lifetime)
+	};
+
+	return rv.get();
+}
+
+rpl::variable<bool> *make_bool(const char *key, Description<bool> &&d) {
+	return make_impl<bool>(key, std::move(d));
+}
+
+rpl::variable<int> *make_int(const char *key, Description<int> &&d) {
+	return make_impl<int>(key, std::move(d));
+}
+
+rpl::variable<QString> *make_qs(const char *key, Description<QString> &&d) {
+	return make_impl<QString>(key, std::move(d));
+}
+
+void update_value(const char *key, bool value) {
+	if (auto it = Options().find(std::string_view(key)); it != Options().end()) {
+		if (const auto *o = std::get_if<Option<bool>>(&it->second)) {
+			*o->v = value;
+		} else {
+			Unexpected("non-bool type in Yukigram::Options::update_value");
+		}
+	}
+}
+
+void init() {
+	for (auto &[key, option] : Options()) {
+		v::match(option, [](Option<bool> &o) {
+			o.init();
+		}, [](Option<int> &o) {
+			o.init();
+		}, [](Option<QString> &o) {
+			o.init();
+		});
+	}
+}
+
+void link() {
+	for (auto &[key, option] : Options()) {
+		v::match(option, [](Option<bool> &o) {
+			o.link();
+		}, [](Option<int> &o) {
+			o.link();
+		}, [](Option<QString> &o) {
+			o.link();
+		});
+	}
+}
+
+
+namespace Valid {
+
+QString BoundedImpl(int lower, int upper, int value) {
+	if (value < lower) {
+		return ktr("input/bounded/lower", { "value", QString::number(value) }, { "lower", QString::number(lower) });
+	} else if (value > upper) {
+		return ktr("input/bounded/upper", { "value", QString::number(value) }, { "upper", QString::number(upper) });
+	}
+	return {};
+}
+
+CheckValid<int> Bounded(int lower, int upper) {
+	return std::bind_front(BoundedImpl, lower, upper);
+}
+
+std::function<rpl::producer<QString>()> BoundedHint(int lower, int upper) {
+	return [=]() {
+		return rktr("input/bounded/hint", { "lower", QString::number(lower) }, { "upper", QString::number(upper) });
+	};
+}
+
+} // namespace Valid
+
+
+} // namespace Yukigram::Options
diff --git a/Telegram/SourceFiles/yukigram/options.h b/Telegram/SourceFiles/yukigram/options.h
new file mode 100644
index 0000000000..4f7ac13e0e
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/options.h
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "rpl/rpl.h"
+
+#include <QStringList>
+#include <string_view>
+
+namespace Yukigram::Options {
+
+template<typename T> using CheckValid = std::function<QString(T)>;
+
+namespace Valid {
+template<typename T> QString Always(T) {
+	return {};
+}
+CheckValid<int> Bounded(int lower, int upper);
+std::function<rpl::producer<QString>()> BoundedHint(int lower, int upper);
+} // namespace Valid
+
+template<typename T> struct Description {
+	QStringList keywords;
+	std::string_view category;
+	bool restartRequired = false;
+	bool mirrorExperimental = false;
+	T defaultValue;
+	std::function<void(T)> link = [](T) {};
+	CheckValid<T> checkValid = Valid::Always<T>;
+	std::function<rpl::producer<QString>()> validHint;
+};
+
+template<typename T> rpl::variable<T> *make(const char *key, Description<T> &&d) {
+	if constexpr (std::is_same_v<T, bool>) {
+		return make_bool(key, std::move(d));
+	} else if constexpr (std::is_same_v<T, int>) {
+		return make_int(key, std::move(d));
+	} else if constexpr (std::is_same_v<T, QString>) {
+		return make_qs(key, std::move(d));
+	}
+}
+rpl::variable<bool> *make_bool(const char *key, Description<bool> &&d);
+rpl::variable<int> *make_int(const char *key, Description<int> &&d);
+rpl::variable<QString> *make_qs(const char *key, Description<QString> &&d);
+
+} // namepace Yukigram::Options
diff --git a/Telegram/SourceFiles/yukigram/options_database.h b/Telegram/SourceFiles/yukigram/options_database.h
new file mode 100644
index 0000000000..df53fcb22d
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/options_database.h
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "yukigram/options.h"
+#include "base/options.h"
+
+#include <map>
+#include <memory>
+#include <optional>
+#include <string_view>
+#include <vector>
+
+namespace Yukigram::Options {
+
+template<typename T> struct Option {
+	std::string_view key;
+	Description<T> d;
+	std::shared_ptr<rpl::variable<T>> v;
+	std::optional<std::shared_ptr<base::options::option<T>>> _base;
+	rpl::lifetime _lifetime;
+
+	void init() {
+		auto &inner = base::options::lookup<T>(std::string(key).c_str());
+		const auto value = inner.value();
+		if (const auto reason = d.checkValid(value); !reason.isEmpty()) {
+			LOG(("Yukigram::Options::Option: option %1: value %2: %3. Resetting to default %4"
+				).arg(key).arg(value).arg(reason).arg(d.defaultValue));
+			*v = d.defaultValue;
+		} else {
+			*v = value;
+		}
+	}
+	void link() {
+		auto &inner = base::options::lookup<T>(std::string(key).c_str());
+		inner.set(v->current());
+		v->value() | rpl::on_next([=, &inner, link = std::move(d.link)](T value) {
+			inner.set(value);
+			link(value);
+		}, _lifetime);
+	}
+};
+
+std::map<std::string_view, std::variant<Option<bool>, Option<int>, Option<QString>>> &Options();
+std::map<std::string_view, std::vector<std::string_view>> &ByCategory();
+
+}
diff --git a/Telegram/SourceFiles/yukigram/options_runtime.h b/Telegram/SourceFiles/yukigram/options_runtime.h
new file mode 100644
index 0000000000..1330c97017
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/options_runtime.h
@@ -0,0 +1,10 @@
+#pragma once
+
+namespace Yukigram::Options {
+
+void update_value(const char *key, bool value);
+
+void init();
+void link();
+
+}
