From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Tue, 12 May 2026 01:21:33 +0300
Subject: [PATCH] (Annoyances) Disable AI IV summaries


diff --git a/Telegram/SourceFiles/iv/iv_rich_page.cpp b/Telegram/SourceFiles/iv/iv_rich_page.cpp
index c9b232ea0f..bead0f18af 100644
--- a/Telegram/SourceFiles/iv/iv_rich_page.cpp
+++ b/Telegram/SourceFiles/iv/iv_rich_page.cpp
@@ -1271,7 +1271,9 @@ void AppendBlock(
 		AdoptAnchor(&parsed.anchorId, &parsed.text);
 		parsed.caption = ParseRichText(data.vcaption(), context);
 		AdoptAnchor(&parsed.anchorId, &parsed.caption);
-		result->push_back(std::move(parsed));
+		if (parsed.caption.text != TextWithEntities{"Cocoon AI Summary"}) {
+			result->push_back(std::move(parsed));
+		}
 	}, [&](const MTPDpageBlockBlockquoteBlocks &data) {
 		auto parsed = MakeBlock(BlockKind::Quote);
 		AppendBlocks(data.vblocks().v, &parsed.blocks, context);
