From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <yuki.utk8g@gmail.com>
Date: Fri, 3 Apr 2026 02:59:31 +0300
Subject: [PATCH] (Minor) tg://nya handler


diff --git a/Telegram/SourceFiles/core/local_url_handlers.cpp b/Telegram/SourceFiles/core/local_url_handlers.cpp
index 5b13c00f1c..595c2c20bd 100644
--- a/Telegram/SourceFiles/core/local_url_handlers.cpp
+++ b/Telegram/SourceFiles/core/local_url_handlers.cpp
@@ -24,6 +24,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "ui/boxes/confirm_box.h"
 #include "ui/boxes/edit_birthday_box.h"
 #include "ui/integration.h"
+#include "yukigram/lang.h"
 #include "payments/payments_non_panel_process.h"
 #include "boxes/peers/edit_peer_info_box.h"
 #include "boxes/share_box.h"
@@ -1700,6 +1701,22 @@ bool ResolveOAuth(
 	return true;
 }
 
+bool ShowNya(
+		Window::SessionController *controller,
+		const Match &match,
+		const QVariant &context) {
+	Ui::Toast::Show(ktr("nya"));
+	return true;
+}
+
+bool ShowWoof(
+		Window::SessionController *controller,
+		const Match &match,
+		const QVariant &context) {
+	Ui::Toast::Show(ktr("woof"));
+	return true;
+}
+
 } // namespace
 
 bool TryRouterForLocalUrl(
@@ -1710,6 +1727,14 @@ bool TryRouterForLocalUrl(
 
 const std::vector<LocalUrlHandler> &LocalUrlHandlers() {
 	static auto Result = std::vector<LocalUrlHandler>{
+		{
+			u"^nya$"_q,
+			ShowNya
+		},
+		{
+			u"^woof$"_q,
+			ShowWoof
+		},
 		{
 			u"^join/?\\?invite=([a-zA-Z0-9\\.\\_\\-]+)(&|$)"_q,
 			JoinGroupByHash
