From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <yuki.utk8g@gmail.com>
Date: Mon, 6 Apr 2026 14:52:44 +0300
Subject: [PATCH] (Shortcut) Forward/Copy/Steal


diff --git a/Telegram/SourceFiles/boxes/peer_list_box.h b/Telegram/SourceFiles/boxes/peer_list_box.h
index eee902d03f..f293251e5a 100644
--- a/Telegram/SourceFiles/boxes/peer_list_box.h
+++ b/Telegram/SourceFiles/boxes/peer_list_box.h
@@ -364,6 +364,9 @@ struct PeerListState;
 
 class PeerListDelegate {
 public:
+	virtual rpl::producer<QString> peerListGetTitle() {
+		Unexpected("PeerListDelegate::peerListGetTitle");
+	}
 	virtual void peerListSetTitle(rpl::producer<QString> title) = 0;
 	virtual void peerListSetAdditionalTitle(rpl::producer<QString> title) = 0;
 	virtual void peerListSetHideEmpty(bool hide) = 0;
@@ -1203,7 +1206,11 @@ public:
 	[[nodiscard]] rpl::producer<int> multiSelectHeightValue() const;
 	[[nodiscard]] rpl::producer<> noSearchSubmits() const;
 
+	rpl::producer<QString> peerListGetTitle() override {
+		return rpl::duplicate(_title);
+	}
 	void peerListSetTitle(rpl::producer<QString> title) override {
+		_title = rpl::duplicate(title);
 		setTitle(std::move(title));
 	}
 	void peerListSetAdditionalTitle(rpl::producer<QString> title) override {
@@ -1266,6 +1273,8 @@ private:
 	void refreshSectionIndex();
 	void updateSectionIndexGeometry();
 
+	rpl::producer<QString> _title;
+
 	object_ptr<Ui::SlideWrap<Ui::MultiSelect>> _select = { nullptr };
 	object_ptr<PeerListSectionIndex> _sectionIndex = { nullptr };
 
diff --git a/Telegram/SourceFiles/boxes/peer_list_controllers.cpp b/Telegram/SourceFiles/boxes/peer_list_controllers.cpp
index 69c507d435..ae0b4f0c12 100644
--- a/Telegram/SourceFiles/boxes/peer_list_controllers.cpp
+++ b/Telegram/SourceFiles/boxes/peer_list_controllers.cpp
@@ -926,6 +926,7 @@ void ChooseRecipientBoxController::rowClicked(not_null<PeerListRow*> row) {
 			});
 		*weak = owned.data();
 		delegate()->peerListUiShow()->showBox(std::move(owned));
+		(*weak)->setTitle(delegate()->peerListGetTitle());
 		return;
 	} else if (const auto monoforum = peer->monoforum()) {
 		const auto weak = std::make_shared<base::weak_qptr<Ui::BoxContent>>();
@@ -1009,6 +1010,7 @@ void ChooseRecipientBoxController::rowClicked(not_null<PeerListRow*> row) {
 			});
 		*weak = owned.data();
 		delegate()->peerListUiShow()->showBox(std::move(owned));
+		(*weak)->setTitle(delegate()->peerListGetTitle());
 		return;
 	}
 	const auto history = peer->owner().history(peer);
diff --git a/Telegram/SourceFiles/core/shortcuts.cpp b/Telegram/SourceFiles/core/shortcuts.cpp
index f2f0982138..7b7b3331a2 100644
--- a/Telegram/SourceFiles/core/shortcuts.cpp
+++ b/Telegram/SourceFiles/core/shortcuts.cpp
@@ -141,6 +141,9 @@ const auto CommandByName = base::flat_map<QString, Command>{
 	{ u"record_round"_q                  , Command::RecordRound },
 	{ u"show_admin_log"_q                , Command::ShowAdminLog },
 	//
+	{ u"fast_forward"_q      , Command::FastForward },
+	{ u"fast_copy"_q         , Command::FastCopy },
+	{ u"fast_steal"_q        , Command::FastSteal },
 };
 
 const base::flat_map<Command, QString> &CommandNames() {
@@ -536,6 +539,10 @@ void Manager::fillDefaults() {
 
 	set(u"ctrl+r"_q, Command::RecordVoice);
 
+	set(u"alt+f"_q, Command::FastForward);
+	set(u"alt+c"_q, Command::FastCopy);
+	set(u"alt+shift+c"_q, Command::FastSteal);
+
 	_defaults = keysCurrents();
 }
 
diff --git a/Telegram/SourceFiles/core/shortcuts.h b/Telegram/SourceFiles/core/shortcuts.h
index c6e50ec09b..5edcc82e3a 100644
--- a/Telegram/SourceFiles/core/shortcuts.h
+++ b/Telegram/SourceFiles/core/shortcuts.h
@@ -89,6 +89,10 @@ enum class Command {
 	SupportScrollToCurrent,
 	SupportHistoryBack,
 	SupportHistoryForward,
+
+	FastForward,
+	FastCopy,
+	FastSteal,
 };
 
 [[maybe_unused]] constexpr auto kShowFolder = {
diff --git a/Telegram/SourceFiles/history/history_inner_widget.cpp b/Telegram/SourceFiles/history/history_inner_widget.cpp
index 4b55eabcd2..4ea6b4183c 100644
--- a/Telegram/SourceFiles/history/history_inner_widget.cpp
+++ b/Telegram/SourceFiles/history/history_inner_widget.cpp
@@ -11,6 +11,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "chat_helpers/stickers_emoji_pack.h"
 #include "core/application.h"
 #include "core/file_utilities.h"
+#include "core/shortcuts.h"
 #include "core/click_handler_types.h"
 #include "core/phone_click_handler.h"
 #include "data/data_chat_participant_status.h"
@@ -508,6 +509,8 @@ HistoryInner::HistoryInner(
 		update();
 	}, lifetime());
 
+	setupShortcuts();
+
 	HistoryView::Reactions::SetupManagerList(
 		_reactionsManager.get(),
 		_reactionsItem.value());
@@ -6389,6 +6392,38 @@ bool CanSendReply(not_null<const HistoryItem*> item) {
 	return true;
 }
 
+void HistoryInner::setupShortcuts() {
+	Shortcuts::Requests(
+	) | rpl::filter([=] {
+		return Ui::AppInFocus()
+			   && Ui::InFocusChain(this)
+			   && !_controller->isLayerShown();
+	}) | rpl::on_next([=](not_null<Shortcuts::Request*> request) {
+		using Command = Shortcuts::Command;
+		request->check(Command::FastForward, 1) && request->handle([=] {
+			auto selectedState = getSelectionState();
+			if (selectedState.count > 0 && selectedState.canForwardCount == selectedState.count) {
+				_widget->forwardSelected(0);
+			}
+			return true;
+		});
+		request->check(Command::FastCopy, 1) && request->handle([=] {
+			auto selectedState = getSelectionState();
+			if (selectedState.count > 0 && selectedState.canForwardCount == selectedState.count) {
+				_widget->forwardSelected(1);
+			}
+			return true;
+		});
+		request->check(Command::FastSteal, 1) && request->handle([=] {
+			auto selectedState = getSelectionState();
+			if (selectedState.count > 0 && selectedState.canForwardCount == selectedState.count) {
+				_widget->forwardSelected(2);
+			}
+			return true;
+		});
+	}, lifetime());
+}
+
 // Accessibility.
 
 std::vector<HistoryView::Element*> HistoryInner::accessibleElements() const {
diff --git a/Telegram/SourceFiles/history/history_inner_widget.h b/Telegram/SourceFiles/history/history_inner_widget.h
index 12cb3eb786..c69bc6a269 100644
--- a/Telegram/SourceFiles/history/history_inner_widget.h
+++ b/Telegram/SourceFiles/history/history_inner_widget.h
@@ -540,6 +540,7 @@ private:
 	void blockSenderAsGroup(FullMsgId itemId);
 	void copySelectedText();
 	void editCaptionUploadLayer(not_null<HistoryItem*> item);
+	void setupShortcuts();
 
 	[[nodiscard]] auto reactionButtonParameters(
 		not_null<const Element*> view,
diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index 1f4fa855f5..4bcce29deb 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -10473,7 +10473,7 @@ bool HistoryWidget::updateCanSendMessage() {
 	return true;
 }
 
-void HistoryWidget::forwardSelected() {
+void HistoryWidget::forwardSelected(int steal) {
 	if (!_list) {
 		return;
 	}
@@ -10482,7 +10482,7 @@ void HistoryWidget::forwardSelected() {
 		if (const auto strong = weak.get()) {
 			strong->clearSelected();
 		}
-	});
+	}, steal);
 }
 
 void HistoryWidget::confirmDeleteSelected() {
diff --git a/Telegram/SourceFiles/history/history_widget.h b/Telegram/SourceFiles/history/history_widget.h
index f08687b23e..1b9465aa2d 100644
--- a/Telegram/SourceFiles/history/history_widget.h
+++ b/Telegram/SourceFiles/history/history_widget.h
@@ -304,7 +304,7 @@ public:
 
 	QPixmap grabForShowAnimation(const Window::SectionSlideParams &params);
 
-	void forwardSelected();
+	void forwardSelected(int steal = 0);
 	void confirmDeleteSelected();
 	void clearSelected();
 
diff --git a/Telegram/SourceFiles/window/window_peer_menu.cpp b/Telegram/SourceFiles/window/window_peer_menu.cpp
index d00215ee43..94a3d4aeb3 100644
--- a/Telegram/SourceFiles/window/window_peer_menu.cpp
+++ b/Telegram/SourceFiles/window/window_peer_menu.cpp
@@ -3026,7 +3026,8 @@ base::weak_qptr<Ui::BoxContent> ShowChooseRecipientBox(
 base::weak_qptr<Ui::BoxContent> ShowForwardMessagesBox(
 		std::shared_ptr<ChatHelpers::Show> show,
 		Data::ForwardDraft &&draft,
-		Fn<void()> &&successCallback) {
+		Fn<void()> &&successCallback,
+		int steal) {
 	const auto session = &show->session();
 	const auto owner = &session->data();
 	const auto itemsList = owner->idsToItems(draft.ids);
@@ -3039,6 +3040,11 @@ base::weak_qptr<Ui::BoxContent> ShowForwardMessagesBox(
 	const auto showForwardOptions = !hasOnlyForcedForwardedInfo
 		&& (!hasRichPage
 			|| HistoryView::Controls::CanHideForwardAuthor(session, itemsList));
+	if (steal > 1 && captionsCount > 0) {
+		draft.options = Data::ForwardOptions::NoNamesAndCaptions;
+	} else if (steal > 0) {
+		draft.options = Data::ForwardOptions::NoSenderNames;
+	}
 	draft.options = HistoryView::Controls::NormalizeForwardOptions(
 		session,
 		itemsList,
@@ -3224,6 +3230,20 @@ base::weak_qptr<Ui::BoxContent> ShowForwardMessagesBox(
 		box->peerListContent()->restoreState(std::move(state));
 	};
 
+	const auto updateTitle = [&](not_null<ListBox*> box) {
+		const auto flags = box->forwardOptions();
+		const auto getTitle = [&] {
+			if (!flags.dropNames) {
+				return "box/forward/forward";
+			} else if (!flags.dropCaptions) {
+				return "box/forward/copy";
+			} else {
+				return "box/forward/steal";
+			}
+		}();
+		box->peerListSetTitle(rktr(getTitle));
+	};
+
 	const auto state = [&] {
 		auto controller = std::make_unique<Controller>(session);
 		const auto controllerRaw = controller.get();
@@ -3272,13 +3292,15 @@ base::weak_qptr<Ui::BoxContent> ShowForwardMessagesBox(
 				== Data::ForwardOptions::NoNamesAndCaptions),
 		});
 		show->showBox(std::move(box));
+		updateTitle(boxRaw);
 		auto state = State{ boxRaw, controllerRaw };
 		return boxRaw->lifetime().make_state<State>(std::move(state));
 	}();
 
 	{ // Chosen a single.
-		auto chosen = [show, draft = std::move(draft)](
+		auto chosen = [show, state, draft = std::move(draft)](
 				not_null<Data::Thread*> thread) mutable {
+			draft.options = state->box->forwardOptionsData();
 			const auto peer = thread->peer();
 			if (peer->isSelf()
 				&& !draft.ids.empty()
@@ -3469,6 +3491,7 @@ base::weak_qptr<Ui::BoxContent> ShowForwardMessagesBox(
 				state->box->forwardOptions(),
 				[=](Ui::ForwardOptions o) {
 					state->box->setForwardOptions(o);
+					updateTitle(state->box);
 				},
 				state->menu->lifetime());
 
@@ -3580,21 +3603,25 @@ base::weak_qptr<Ui::BoxContent> ShowForwardMessagesBox(
 base::weak_qptr<Ui::BoxContent> ShowForwardMessagesBox(
 		not_null<Window::SessionNavigation*> navigation,
 		Data::ForwardDraft &&draft,
-		Fn<void()> &&successCallback) {
+		Fn<void()> &&successCallback,
+		int steal) {
 	return ShowForwardMessagesBox(
 		navigation->uiShow(),
 		std::move(draft),
-		std::move(successCallback));
+		std::move(successCallback),
+		steal);
 }
 
 base::weak_qptr<Ui::BoxContent> ShowForwardMessagesBox(
 		not_null<Window::SessionNavigation*> navigation,
 		MessageIdsList &&items,
-		Fn<void()> &&successCallback) {
+		Fn<void()> &&successCallback,
+		int steal) {
 	return ShowForwardMessagesBox(
 		navigation,
 		Data::ForwardDraft{ .ids = std::move(items) },
-		std::move(successCallback));
+		std::move(successCallback),
+		steal);
 }
 
 base::weak_qptr<Ui::BoxContent> ShowShareGameBox(
diff --git a/Telegram/SourceFiles/window/window_peer_menu.h b/Telegram/SourceFiles/window/window_peer_menu.h
index 8664c91dbb..35fc31db43 100644
--- a/Telegram/SourceFiles/window/window_peer_menu.h
+++ b/Telegram/SourceFiles/window/window_peer_menu.h
@@ -206,15 +206,18 @@ base::weak_qptr<Ui::BoxContent> ShowChooseRecipientBox(
 base::weak_qptr<Ui::BoxContent> ShowForwardMessagesBox(
 	std::shared_ptr<ChatHelpers::Show> show,
 	Data::ForwardDraft &&draft,
-	Fn<void()> &&successCallback = nullptr);
+	Fn<void()> &&successCallback = nullptr,
+	int steal = 0);
 base::weak_qptr<Ui::BoxContent> ShowForwardMessagesBox(
 	not_null<Window::SessionNavigation*> navigation,
 	Data::ForwardDraft &&draft,
-	Fn<void()> &&successCallback = nullptr);
+	Fn<void()> &&successCallback = nullptr,
+	int steal = 0);
 base::weak_qptr<Ui::BoxContent> ShowForwardMessagesBox(
 	not_null<Window::SessionNavigation*> navigation,
 	MessageIdsList &&items,
-	Fn<void()> &&successCallback = nullptr);
+	Fn<void()> &&successCallback = nullptr,
+	int steal = 0);
 base::weak_qptr<Ui::BoxContent> ShowShareUrlBox(
 	not_null<Window::SessionNavigation*> navigation,
 	const QString &url,
