From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <yuki.utk8g@gmail.com>
Date: Wed, 15 Apr 2026 22:33:48 +0300
Subject: [PATCH] (Shortcut) Hide similar channels on join message by default


diff --git a/Telegram/SourceFiles/apiwrap.cpp b/Telegram/SourceFiles/apiwrap.cpp
index 857f8643c9..0f4a5f6d71 100644
--- a/Telegram/SourceFiles/apiwrap.cpp
+++ b/Telegram/SourceFiles/apiwrap.cpp
@@ -1894,7 +1894,6 @@ void ApiWrap::joinChannel(not_null<ChannelData*> channel) {
 
 		using Flag = ChannelDataFlag;
 		chatParticipants().loadSimilarPeers(channel);
-		channel->setFlags(channel->flags() | Flag::SimilarExpanded);
 	}
 }
 
