From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <yuki.utk8g@gmail.com>
Date: Fri, 3 Apr 2026 03:07:17 +0300
Subject: [PATCH] (Message)(Opt) Disable animoji


diff --git a/Telegram/SourceFiles/history/view/history_view_element.cpp b/Telegram/SourceFiles/history/view/history_view_element.cpp
index 70bf55cd47..b645b33833 100644
--- a/Telegram/SourceFiles/history/view/history_view_element.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_element.cpp
@@ -46,6 +46,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "payments/payments_reaction_process.h" // TryAddingPaidReaction.
 #include "window/window_session_controller.h"
 #include "window/section_widget.h"
+#include "yukigram/settings/disable_animoji.h"
 #include "ui/chat/chat_style.h"
 #include "ui/effects/glare.h"
 #include "ui/effects/path_shift_gradient.h"
@@ -1616,7 +1617,7 @@ void Element::refreshMedia(Element *replacing) {
 		const auto emoji = isolatedEmoji();
 		const auto emojiStickers = &history()->session().emojiStickersPack();
 		const auto skipPremiumEffect = false;
-		if (const auto sticker = emojiStickers->stickerForEmoji(emoji)) {
+		if (const auto sticker = emojiStickers->stickerForEmoji(emoji); sticker && (!Yukigram::Settings::DisableAnimoji->current())) {
 			auto content = std::make_unique<Sticker>(
 				this,
 				sticker.document,
diff --git a/Telegram/SourceFiles/yukigram/settings/disable_animoji.cpp b/Telegram/SourceFiles/yukigram/settings/disable_animoji.cpp
new file mode 100644
index 0000000000..8f63c901e0
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/disable_animoji.cpp
@@ -0,0 +1,14 @@
+#include "yukigram/settings/disable_animoji.h"
+#include "yukigram/options.h"
+
+namespace Yukigram::Settings {
+
+rpl::variable<bool> *DisableAnimoji = Yukigram::Options::make<bool>(kOptionDisableAnimoji, {
+	.keywords = { u"yukigram"_q, u"animoji"_q, u"emoji"_q },
+	.category = "messages",
+	.restartRequired = true,
+});
+
+const char kOptionDisableAnimoji[] = "disable-animoji";
+
+}
diff --git a/Telegram/SourceFiles/yukigram/settings/disable_animoji.h b/Telegram/SourceFiles/yukigram/settings/disable_animoji.h
new file mode 100644
index 0000000000..27322b07a0
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/disable_animoji.h
@@ -0,0 +1,5 @@
+#include "rpl/rpl.h"
+namespace Yukigram::Settings {
+extern rpl::variable<bool> *DisableAnimoji;
+extern const char kOptionDisableAnimoji[];
+}
