From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <yuki.utk8g@gmail.com>
Date: Fri, 3 Apr 2026 03:16:20 +0300
Subject: [PATCH] (Interface)(Opt) Single-column layout


diff --git a/Telegram/SourceFiles/mainwidget.cpp b/Telegram/SourceFiles/mainwidget.cpp
index 78554d8826..3b23c195ad 100644
--- a/Telegram/SourceFiles/mainwidget.cpp
+++ b/Telegram/SourceFiles/mainwidget.cpp
@@ -36,6 +36,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "ui/text/text_utilities.h"
 #include "ui/toast/toast.h"
 #include "ui/ui_utility.h"
+#include "yukigram/settings/force_mobile_layout.h"
 #include "window/window_connecting_widget.h"
 #include "window/window_top_bar_wrap.h"
 #include "window/notifications_manager.h"
@@ -337,6 +338,7 @@ MainWidget::MainWidget(
 		});
 	};
 	rpl::merge(
+		Yukigram::Settings::ForceMobileLayout->changes() | rpl::to_empty,
 		Core::App().settings().dialogsWithChatWidthRatioChanges(
 		) | filter(true) | rpl::to_empty,
 		Core::App().settings().dialogsNoChatWidthRatioChanges(
@@ -2899,7 +2901,7 @@ void MainWidget::updateWindowAdaptiveLayout() {
 
 	// Check if we are in a single-column layout in a wide enough window
 	// for the normal layout. If so, switch to the normal layout.
-	if (layout.windowLayout == Window::Adaptive::WindowLayout::OneColumn) {
+	if ((!Yukigram::Settings::ForceMobileLayout->current()) && layout.windowLayout == Window::Adaptive::WindowLayout::OneColumn) {
 		auto chatWidth = layout.chatWidth;
 		//if (session().settings().tabbedSelectorSectionEnabled()
 		//	&& chatWidth >= _history->minimalWidthForTabbedSelectorSection()) {
diff --git a/Telegram/SourceFiles/window/window_adaptive.cpp b/Telegram/SourceFiles/window/window_adaptive.cpp
index 784164db99..198efb8f88 100644
--- a/Telegram/SourceFiles/window/window_adaptive.cpp
+++ b/Telegram/SourceFiles/window/window_adaptive.cpp
@@ -8,6 +8,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "window/window_adaptive.h"
 
 #include "history/history_item.h"
+#include "yukigram/settings/force_mobile_layout.h"
 #include "data/data_media_types.h"
 #include "data/data_session.h"
 #include "core/application.h"
diff --git a/Telegram/SourceFiles/window/window_session_controller.cpp b/Telegram/SourceFiles/window/window_session_controller.cpp
index 9654449223..f4e7412d24 100644
--- a/Telegram/SourceFiles/window/window_session_controller.cpp
+++ b/Telegram/SourceFiles/window/window_session_controller.cpp
@@ -89,6 +89,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "ui/effects/message_sending_animation_controller.h"
 #include "ui/style/style_palette_colorizer.h"
 #include "ui/toast/toast.h"
+#include "yukigram/settings/force_mobile_layout.h"
 #include "calls/calls_instance.h" // Core::App().calls().inCall().
 #include "calls/group/calls_group_call.h"
 #include "calls/group/calls_group_common.h"
@@ -2597,6 +2598,7 @@ auto SessionController::computeColumnLayout() const -> ColumnLayout {
 	auto dialogsWidth = 0, chatWidth = 0, thirdWidth = 0;
 
 	auto useOneColumnLayout = [&] {
+		if (Yukigram::Settings::ForceMobileLayout->current()) return true;
 		auto minimalNormal = st::columnMinimalWidthLeft
 			+ st::columnMinimalWidthMain;
 		if (_hasDialogs && bodyWidth < minimalNormal) {
diff --git a/Telegram/SourceFiles/yukigram/settings/force_mobile_layout.cpp b/Telegram/SourceFiles/yukigram/settings/force_mobile_layout.cpp
new file mode 100644
index 0000000000..c9f2bb2243
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/force_mobile_layout.cpp
@@ -0,0 +1,13 @@
+#include "yukigram/settings/force_mobile_layout.h"
+#include "yukigram/options.h"
+
+namespace Yukigram::Settings {
+
+rpl::variable<bool> *ForceMobileLayout = Yukigram::Options::make<bool>(kOptionForceMobileLayout, {
+	.keywords = { u"yukigram"_q, u"layout"_q, u"small"_q, u"phone"_q, u"mobile"_q },
+	.category = "interface",
+});
+
+const char kOptionForceMobileLayout[] = "force-mobile-layout";
+
+}
diff --git a/Telegram/SourceFiles/yukigram/settings/force_mobile_layout.h b/Telegram/SourceFiles/yukigram/settings/force_mobile_layout.h
new file mode 100644
index 0000000000..8220914012
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/force_mobile_layout.h
@@ -0,0 +1,5 @@
+#include "rpl/rpl.h"
+namespace Yukigram::Settings {
+extern rpl::variable<bool> *ForceMobileLayout;
+extern const char kOptionForceMobileLayout[];
+}
