From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <yuki.utk8g@gmail.com>
Date: Fri, 3 Apr 2026 04:45:03 +0300
Subject: [PATCH] (Compose)(Opt) Show scheduled messages button


diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index c44515c3eb..fcd6cd49ce 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -36,6 +36,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "core/mime_type.h"
 #include "history/view/history_view_draw_to_reply.h"
 #include "history/view/controls/history_view_rich_draft_preview.h"
+#include "yukigram/settings/always_show_scheduled.h"
 #include "ui/emoji_config.h"
 #include "ui/chat/attach/attach_prepare.h"
 #include "ui/chat/choose_theme_controller.h"
@@ -3734,7 +3735,7 @@ void HistoryWidget::setupScheduledToggle() {
 }
 
 void HistoryWidget::refreshScheduledToggle() {
-	const auto has = _history
+	const auto has = Yukigram::Settings::AlwaysShowScheduled->current() || _history
 		&& _canSendMessages
 		&& (session().scheduledMessages().count(_history) > 0);
 	if (!_scheduled && has) {
diff --git a/Telegram/SourceFiles/yukigram/settings/always_show_scheduled.cpp b/Telegram/SourceFiles/yukigram/settings/always_show_scheduled.cpp
new file mode 100644
index 0000000000..6962836dd6
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/always_show_scheduled.cpp
@@ -0,0 +1,13 @@
+#include "yukigram/settings/always_show_scheduled.h"
+#include "yukigram/options.h"
+
+namespace Yukigram::Settings {
+
+rpl::variable<bool> *AlwaysShowScheduled = Yukigram::Options::make<bool>(kOptionAlwaysShowScheduled, {
+	.keywords = { u"yukigram"_q, u"scheduled"_q, u"delay"_q },
+	.category = "compose",
+});
+
+const char kOptionAlwaysShowScheduled[] = "always-show-scheduled";
+
+}
diff --git a/Telegram/SourceFiles/yukigram/settings/always_show_scheduled.h b/Telegram/SourceFiles/yukigram/settings/always_show_scheduled.h
new file mode 100644
index 0000000000..910bfebadb
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/always_show_scheduled.h
@@ -0,0 +1,5 @@
+#include "rpl/rpl.h"
+namespace Yukigram::Settings {
+extern rpl::variable<bool> *AlwaysShowScheduled;
+extern const char kOptionAlwaysShowScheduled[];
+}
