From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Thu, 7 May 2026 11:11:58 +0300
Subject: [PATCH] (Style)(Opt) Hide message tail


diff --git a/Telegram/SourceFiles/history/view/history_view_message.cpp b/Telegram/SourceFiles/history/view/history_view_message.cpp
index 5ec178e240..6b378ddba0 100644
--- a/Telegram/SourceFiles/history/view/history_view_message.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_message.cpp
@@ -33,6 +33,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "history/history.h"
 #include "iv/iv_instance.h"
 #include "iv/iv_rich_page.h"
+#include "yukigram/settings/hide_message_tail.h"
 #include "boxes/premium_preview_box.h"
 #include "boxes/share_box.h"
 #include "boxes/peers/tag_info_box.h"
@@ -6304,6 +6305,7 @@ Ui::BubbleRounding Message::countMessageRounding() const {
 	const auto item = data();
 	const auto keyboard = item->inlineReplyKeyboard();
 	const auto skipTail = smallBottom
+		|| Yukigram::Settings::HideMessageTail->current()
 		|| (media && media->skipBubbleTail())
 		|| (keyboard != nullptr)
 		|| item->isFakeAboutView()
diff --git a/Telegram/SourceFiles/yukigram/settings/hide_message_tail.cpp b/Telegram/SourceFiles/yukigram/settings/hide_message_tail.cpp
new file mode 100644
index 0000000000..abe190b827
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/hide_message_tail.cpp
@@ -0,0 +1,14 @@
+#include "yukigram/settings/hide_message_tail.h"
+#include "yukigram/options.h"
+
+namespace Yukigram::Settings {
+
+rpl::variable<bool> *HideMessageTail = Yukigram::Options::make<bool>(kOptionHideMessageTail, {
+	.keywords = { u"yukigram"_q, u"message"_q, u"bubble"_q, u"tail"_q },
+	.category = "style",
+	.restartRequired = true,
+});
+
+const char kOptionHideMessageTail[] = "hide-message-tail";
+
+}
diff --git a/Telegram/SourceFiles/yukigram/settings/hide_message_tail.h b/Telegram/SourceFiles/yukigram/settings/hide_message_tail.h
new file mode 100644
index 0000000000..d7d4b4e793
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/hide_message_tail.h
@@ -0,0 +1,5 @@
+#include "rpl/rpl.h"
+namespace Yukigram::Settings {
+extern rpl::variable<bool> *HideMessageTail;
+extern const char kOptionHideMessageTail[];
+}
