From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Thu, 21 May 2026 05:40:28 +0300
Subject: [PATCH] (Fix) Webview socket path


diff --git a/Telegram/SourceFiles/platform/linux/specific_linux.cpp b/Telegram/SourceFiles/platform/linux/specific_linux.cpp
index 0e8c317294..82f70152c1 100644
--- a/Telegram/SourceFiles/platform/linux/specific_linux.cpp
+++ b/Telegram/SourceFiles/platform/linux/specific_linux.cpp
@@ -30,6 +30,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include <QtWidgets/QApplication>
 #include <QtWidgets/QSystemTrayIcon>
 #include <QtGui/QDesktopServices>
+#include <QtCore/QRandomGenerator>
 #include <QtCore/QStandardPaths>
 #include <QtCore/QProcess>
 
@@ -769,9 +770,10 @@ void start() {
 	GLib::set_prgname(cExeName().toStdString());
 	GLib::set_application_name(AppName.data());
 
-	Webview::WebKitGTK::SetSocketPath(u"%1/%2-%3-webview-{}"_q.arg(
+	Webview::WebKitGTK::SetSocketPath(u"%1/%2-%3-%4-webview-{}"_q.arg(
 		QDir::tempPath(),
 		HashForSocketPath(),
+		QString::number(QRandomGenerator::global()->generate(), 36),
 		u"TD"_q).toStdString());
 
 	InstallLauncher();
