From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Sun, 24 May 2026 16:33:21 +0300
Subject: [PATCH] (Major) Split Stickers and GIFs chat permission toggle


diff --git a/Telegram/Resources/langs/lang.strings b/Telegram/Resources/langs/lang.strings
index eceb514767..b5b89385d3 100644
--- a/Telegram/Resources/langs/lang.strings
+++ b/Telegram/Resources/langs/lang.strings
@@ -7033,6 +7033,10 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 "lng_admin_log_banned_send_voice_messages" = "Send voice messages";
 "lng_admin_log_banned_send_video_messages" = "Send video messages";
 "lng_admin_log_banned_send_stickers" = "Send stickers & GIFs";
+"lng_admin_log_banned_send_stickers_only" = "Send stickers";
+"lng_admin_log_banned_send_gifs" = "Send GIFs";
+"lng_admin_log_banned_send_inline" = "Send inline bots";
+"lng_admin_log_banned_send_games" = "Send games";
 "lng_admin_log_banned_embed_links" = "Embed links";
 "lng_admin_log_banned_send_polls" = "Send polls";
 "lng_admin_log_banned_send_reactions" = "Send reactions";
diff --git a/Telegram/SourceFiles/boxes/peers/edit_peer_permissions_box.cpp b/Telegram/SourceFiles/boxes/peers/edit_peer_permissions_box.cpp
index 3b2fe36f99..e27fff6543 100644
--- a/Telegram/SourceFiles/boxes/peers/edit_peer_permissions_box.cpp
+++ b/Telegram/SourceFiles/boxes/peers/edit_peer_permissions_box.cpp
@@ -8,6 +8,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "boxes/peers/edit_peer_permissions_box.h"
 
 #include "lang/lang_keys.h"
+#include "yukigram/lang.h"
 #include "history/admin_log/history_admin_log_filter.h"
 #include "core/ui_integration.h"
 #include "data/stickers/data_custom_emoji.h"
@@ -85,10 +86,10 @@ constexpr auto kDefaultChargeStars = 10;
 		{ Flag::SendMusic, tr::lng_rights_chat_music(tr::now) },
 		{ Flag::SendVoiceMessages, tr::lng_rights_chat_voice_messages(tr::now) },
 		{ Flag::SendFiles, tr::lng_rights_chat_files(tr::now) },
-		{ Flag::SendStickers
-			| Flag::SendGifs
-			| Flag::SendGames
-			| Flag::SendInline, tr::lng_rights_chat_stickers(tr::now) },
+		{ Flag::SendStickers, ktr("box/edit-peer-permissions/rights/stickers") },
+		{ Flag::SendGifs, ktr("box/edit-peer-permissions/rights/gifs") },
+		{ Flag::SendGames, ktr("box/edit-peer-permissions/rights/games") },
+		{ Flag::SendInline, ktr("box/edit-peer-permissions/rights/inline") },
 		{ Flag::EmbedLinks, tr::lng_rights_chat_send_links(tr::now) },
 		{ Flag::SendPolls, tr::lng_rights_chat_send_polls(tr::now) },
 		{ Flag::SendReactions, tr::lng_rights_chat_send_reactions(tr::now) },
@@ -290,18 +291,6 @@ auto Dependencies(ChatRestrictions)
 	using Flag = ChatRestriction;
 
 	return {
-		// stickers <-> gifs
-		{ Flag::SendGifs, Flag::SendStickers },
-		{ Flag::SendStickers, Flag::SendGifs },
-
-		// stickers <-> games
-		{ Flag::SendGames, Flag::SendStickers },
-		{ Flag::SendStickers, Flag::SendGames },
-
-		// stickers <-> inline
-		{ Flag::SendInline, Flag::SendStickers },
-		{ Flag::SendStickers, Flag::SendInline },
-
 		// embed_links -> send_plain
 		{ Flag::EmbedLinks, Flag::SendOther },
 
@@ -1471,14 +1460,6 @@ ChatAdminRights DisabledByDefaultRestrictions(not_null<PeerData*> peer) {
 ChatRestrictions FixDependentRestrictions(ChatRestrictions restrictions) {
 	const auto &dependencies = Dependencies(restrictions);
 
-	// Fix iOS bug of saving send_inline like embed_links.
-	// We copy send_stickers to send_inline.
-	if (restrictions & ChatRestriction::SendStickers) {
-		restrictions |= ChatRestriction::SendInline;
-	} else {
-		restrictions &= ~ChatRestriction::SendInline;
-	}
-
 	// Apply the strictest.
 	const auto fixOne = [&] {
 		for (const auto &[first, second] : dependencies) {
diff --git a/Telegram/SourceFiles/history/admin_log/history_admin_log_item.cpp b/Telegram/SourceFiles/history/admin_log/history_admin_log_item.cpp
index 250eceb7a4..df5eeef392 100644
--- a/Telegram/SourceFiles/history/admin_log/history_admin_log_item.cpp
+++ b/Telegram/SourceFiles/history/admin_log/history_admin_log_item.cpp
@@ -336,10 +336,10 @@ QString GeneratePermissionsChangeText(
 		{
 			Flag::SendVideoMessages,
 			tr::lng_admin_log_banned_send_video_messages },
-		{ Flag::SendStickers
-			| Flag::SendGifs
-			| Flag::SendInline
-			| Flag::SendGames, tr::lng_admin_log_banned_send_stickers },
+		{ Flag::SendStickers, tr::lng_admin_log_banned_send_stickers_only },
+		{ Flag::SendGifs, tr::lng_admin_log_banned_send_gifs },
+		{ Flag::SendInline, tr::lng_admin_log_banned_send_inline },
+		{ Flag::SendGames, tr::lng_admin_log_banned_send_games },
 		{ Flag::EmbedLinks, tr::lng_admin_log_banned_embed_links },
 		{ Flag::SendReactions, tr::lng_admin_log_banned_send_reactions },
 		{ Flag::SendPolls, tr::lng_admin_log_banned_send_polls },
diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index 7fba0bcae3..2ad60c9917 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -9679,7 +9679,7 @@ bool HistoryWidget::sendExistingDocument(
 	const auto ephemeralReply = session().ephemeralMessages()
 		.isEphemeralBotReply(messageToSend.action.replyTo.messageId);
 	const auto error = (_peer && !ephemeralReply)
-		? Data::RestrictionError(_peer, ChatRestriction::SendStickers)
+		? Data::RestrictionError(_peer, document->isGifv() ? ChatRestriction::SendGifs : ChatRestriction::SendStickers)
 		: Data::SendError();
 	if (error) {
 		Data::ShowSendErrorToast(controller(), _peer, error);
diff --git a/Telegram/SourceFiles/history/view/history_view_chat_section.cpp b/Telegram/SourceFiles/history/view/history_view_chat_section.cpp
index b8ae6dfa02..41956c1b75 100644
--- a/Telegram/SourceFiles/history/view/history_view_chat_section.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_chat_section.cpp
@@ -74,6 +74,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "data/data_saved_messages.h"
 #include "data/data_saved_sublist.h"
 #include "data/data_session.h"
+#include "data/data_document.h"
 #include "data/data_user.h"
 #include "data/data_chat.h"
 #include "data/data_channel.h"
@@ -1921,7 +1922,7 @@ bool ChatWidget::sendExistingDocument(
 	const auto ephemeralReply = session().ephemeralMessages()
 		.isEphemeralBotReply(messageToSend.action.replyTo.messageId);
 	const auto error = !ephemeralReply
-		? Data::RestrictionError(_peer, ChatRestriction::SendStickers)
+		? Data::RestrictionError(_peer, document->isGifv() ? ChatRestriction::SendGifs : ChatRestriction::SendStickers)
 		: Data::SendError();
 	if (error) {
 		Data::ShowSendErrorToast(controller(), _peer, error);
diff --git a/Telegram/SourceFiles/history/view/history_view_scheduled_section.cpp b/Telegram/SourceFiles/history/view/history_view_scheduled_section.cpp
index f3c2e537bf..83e0ba90d9 100644
--- a/Telegram/SourceFiles/history/view/history_view_scheduled_section.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_scheduled_section.cpp
@@ -850,7 +850,7 @@ bool ScheduledWidget::sendExistingDocument(
 		Api::MessageToSend messageToSend) {
 	const auto error = Data::RestrictionError(
 		_history->peer,
-		ChatRestriction::SendStickers);
+		document->isGifv() ? ChatRestriction::SendGifs : ChatRestriction::SendStickers);
 	if (error) {
 		Data::ShowSendErrorToast(controller(), _history->peer, error);
 		return false;
diff --git a/Telegram/SourceFiles/media/stories/media_stories_reply.cpp b/Telegram/SourceFiles/media/stories/media_stories_reply.cpp
index c1fae9a2ee..0d2ca8e873 100644
--- a/Telegram/SourceFiles/media/stories/media_stories_reply.cpp
+++ b/Telegram/SourceFiles/media/stories/media_stories_reply.cpp
@@ -375,7 +375,7 @@ bool ReplyArea::sendExistingDocument(
 	const auto show = _controller->uiShow();
 	const auto error = Data::RestrictionError(
 		_data.peer,
-		ChatRestriction::SendStickers);
+		document->isGifv() ? ChatRestriction::SendGifs : ChatRestriction::SendStickers);
 	if (error) {
 		Data::ShowSendErrorToast(show, _data.peer, error);
 		return false;
