From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Wed, 27 May 2026 19:11:30 +0000
Subject: [PATCH] (Chat)(Opt) Hide bottom button bar


diff --git a/Telegram/SourceFiles/history/history_widget.cpp b/Telegram/SourceFiles/history/history_widget.cpp
index 9310c5f5a0..6befdf30aa 100644
--- a/Telegram/SourceFiles/history/history_widget.cpp
+++ b/Telegram/SourceFiles/history/history_widget.cpp
@@ -104,6 +104,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "data/data_premium_limits.h" // Data::PremiumLimits.
 #include "data/stickers/data_stickers.h"
 #include "data/stickers/data_custom_emoji.h"
+#include "yukigram/settings/hide_bottom_bar.h"
 #include "history/history.h"
 #include "history/history_item.h"
 #include "history/history_item_helpers.h" // GetErrorForSending.
@@ -559,6 +560,12 @@ HistoryWidget::HistoryWidget(
 		updateControlsGeometry();
 	}, lifetime());
 
+	Yukigram::Settings::HideBottomBar->changes(
+	) | rpl::on_next([=] {
+		updateControlsVisibility();
+		updateControlsGeometry();
+	}, lifetime());
+
 	_fieldChatStyle = InitMessageField(controller, _field, [=](
 			not_null<DocumentData*> document) {
 		if (_peer && Data::AllowEmojiWithoutPremium(_peer, document)) {
@@ -6639,6 +6646,7 @@ bool HistoryWidget::isBotStart() const {
 	const auto user = _peer ? _peer->asUser() : nullptr;
 	if (!user
 		|| !user->isBot()
+		// || (user->isBlocked() && Yukigram::Settings::HideBottomBar->current())
 		|| !_canSendMessages) {
 		return false;
 	} else if (!user->botInfo->startToken.isEmpty()) {
@@ -6654,10 +6662,12 @@ bool HistoryWidget::isReportMessages() const {
 }
 
 bool HistoryWidget::isBlocked() const {
+	// if (Yukigram::Settings::HideBottomBar->current()) return false;
 	return _peer && _peer->isUser() && _peer->asUser()->isBlocked();
 }
 
 bool HistoryWidget::isJoinChannel() const {
+	if (Yukigram::Settings::HideBottomBar->current()) return false;
 	if (const auto channel = _peer ? _peer->asChannel() : nullptr) {
 		return !channel->amIn() && !channel->isMonoforum() && !_canSendMessages;
 	}
@@ -6669,6 +6679,7 @@ bool HistoryWidget::isChoosingTheme() const {
 }
 
 bool HistoryWidget::isMuteUnmute() const {
+	if (Yukigram::Settings::HideBottomBar->current()) return false;
 	return _peer
 		&& ((_peer->isBroadcast() && !_peer->asChannel()->canPostMessages())
 			|| (_peer->isGigagroup() && !Data::CanSendAnything(_peer))
diff --git a/Telegram/SourceFiles/history/view/controls/history_view_bottom_controls.cpp b/Telegram/SourceFiles/history/view/controls/history_view_bottom_controls.cpp
index c34d06a26e..7d3f3057b5 100644
--- a/Telegram/SourceFiles/history/view/controls/history_view_bottom_controls.cpp
+++ b/Telegram/SourceFiles/history/view/controls/history_view_bottom_controls.cpp
@@ -21,6 +21,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "data/data_session.h"
 #include "data/data_user.h"
 #include "data/notify/data_notify_settings.h"
+#include "yukigram/settings/hide_bottom_bar.h"
 #include "history/history.h"
 #include "history/history_item_helpers.h"
 #include "history/view/controls/compose_controls_common.h"
@@ -544,6 +545,7 @@ bool BottomControls::isJoinChannel() const {
 		return false;
 	}
 	if (const auto channel = _peer->asChannel()) {
+		if (Yukigram::Settings::HideBottomBar->current()) return false;
 		return !channel->amIn() && !channel->isMonoforum();
 	}
 	return false;
@@ -560,6 +562,7 @@ bool BottomControls::isJoinGroup() const {
 	const auto canSend = !channel->isForum()
 		? Data::CanSendAnything(channel)
 		: (_topic && Data::CanSendAnything(_topic));
+	if (Yukigram::Settings::HideBottomBar->current()) return false;
 	return !channel->amIn() && !canSend;
 }
 
@@ -567,6 +570,7 @@ bool BottomControls::isMuteUnmute() const {
 	if (_mode != BottomControlsMode::History) {
 		return false;
 	}
+	if (Yukigram::Settings::HideBottomBar->current()) return false;
 	return (_peer->isBroadcast() && !_peer->asChannel()->canPostMessages())
 		|| (_peer->isGigagroup() && !Data::CanSendAnything(_peer))
 		|| _peer->isRepliesChat()
diff --git a/Telegram/SourceFiles/yukigram/settings/hide_bottom_bar.cpp b/Telegram/SourceFiles/yukigram/settings/hide_bottom_bar.cpp
new file mode 100644
index 0000000000..a48a380230
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/hide_bottom_bar.cpp
@@ -0,0 +1,13 @@
+#include "yukigram/settings/hide_bottom_bar.h"
+#include "yukigram/options.h"
+
+namespace Yukigram::Settings {
+
+rpl::variable<bool> *HideBottomBar = Yukigram::Options::make<bool>(kOptionHideBottomBar, {
+	.keywords = { u"yukigram"_q, u"bottom"_q, u"bar"_q, u"join"_q, u"mute"_q, u"unmute"_q, u"gift"_q, u"direct"_q },
+	.category = "chat",
+});
+
+const char kOptionHideBottomBar[] = "hide-bottom-bar";
+
+}
diff --git a/Telegram/SourceFiles/yukigram/settings/hide_bottom_bar.h b/Telegram/SourceFiles/yukigram/settings/hide_bottom_bar.h
new file mode 100644
index 0000000000..4efd4816a1
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/hide_bottom_bar.h
@@ -0,0 +1,5 @@
+#include "rpl/rpl.h"
+namespace Yukigram::Settings {
+extern rpl::variable<bool> *HideBottomBar;
+extern const char kOptionHideBottomBar[];
+}
