From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Wed, 27 May 2026 20:18:10 +0000
Subject: [PATCH] (Fix) Non-admin monoforum button in menu


diff --git a/Telegram/SourceFiles/window/window_peer_menu.cpp b/Telegram/SourceFiles/window/window_peer_menu.cpp
index 0ab624f223..6d3f0d48e2 100644
--- a/Telegram/SourceFiles/window/window_peer_menu.cpp
+++ b/Telegram/SourceFiles/window/window_peer_menu.cpp
@@ -1027,7 +1027,7 @@ void Filler::addDirectMessages() {
 		return;
 	}
 	const auto monoforum = channel->broadcastMonoforum();
-	if (!monoforum || !monoforum->amMonoforumAdmin()) {
+	if (!monoforum) {
 		return;
 	}
 	const auto navigation = _controller;
