From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Wed, 27 May 2026 21:29:45 +0000
Subject: [PATCH] (Style)(Opt) Old-style spoilers


diff --git a/Telegram/SourceFiles/history/view/media/history_view_media.cpp b/Telegram/SourceFiles/history/view/media/history_view_media.cpp
index a116d6ac20..63f2eb2d37 100644
--- a/Telegram/SourceFiles/history/view/media/history_view_media.cpp
+++ b/Telegram/SourceFiles/history/view/media/history_view_media.cpp
@@ -21,6 +21,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "data/data_session.h"
 #include "data/data_web_page.h"
 #include "lang/lang_keys.h"
+#include "yukigram/lang.h"
 #include "ui/item_text_options.h"
 #include "ui/chat/chat_style.h"
 #include "ui/chat/message_bubble.h"
@@ -30,6 +31,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "ui/painter.h"
 #include "ui/power_saving.h"
 #include "ui/text/text_utilities.h"
+#include "yukigram/settings/old_style_spoilers.h"
 #include "core/ui_integration.h"
 #include "styles/style_chat.h"
 #include "styles/style_chat_helpers.h"
@@ -361,6 +363,18 @@ void Media::fillImageSpoiler(
 		Ui::DefaultImageSpoiler().frame(
 			spoiler->animation->index(context.now, pausedSpoiler)),
 		spoiler->cornerCache);
+if (Yukigram::Settings::OldStyleSpoilers->current()) {
+	const auto center = rect.topRight() + QPoint(-24, 24);
+	const auto iconRect = QRect(rect.topRight() - QPoint(48, 0), QSize(48, 48));
+	const auto radius = 18;
+	p.setPen(Qt::NoPen);
+	p.setBrush(st::msgDateImgBg->c);
+	{
+		auto hq = PainterHighQualityEnabler(p);
+		p.drawEllipse(center, radius, radius);
+	}
+	st::menuIconSpoiler.paintInCenter(p, iconRect);
+}
 }
 
 void Media::drawSpoilerTag(
diff --git a/Telegram/SourceFiles/yukigram/settings/old_style_spoilers.cpp b/Telegram/SourceFiles/yukigram/settings/old_style_spoilers.cpp
new file mode 100644
index 0000000000..cb47e524e7
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/old_style_spoilers.cpp
@@ -0,0 +1,18 @@
+#include "yukigram/settings/old_style_spoilers.h"
+#include "yukigram/options.h"
+
+namespace Ui {
+extern bool Yukigram_OldStyleSpoilers;
+}
+
+namespace Yukigram::Settings {
+
+rpl::variable<bool> *OldStyleSpoilers = Yukigram::Options::make<bool>(kOptionOldStyleSpoilers, {
+	.keywords = { u"yukigram"_q, u"spoilers"_q, u"mess"_q },
+	.category = "style",
+	.link = [](bool value) { Ui::Yukigram_OldStyleSpoilers = value; },
+});
+
+const char kOptionOldStyleSpoilers[] = "old-style-spoilers";
+
+}
diff --git a/Telegram/SourceFiles/yukigram/settings/old_style_spoilers.h b/Telegram/SourceFiles/yukigram/settings/old_style_spoilers.h
new file mode 100644
index 0000000000..4a84202013
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/old_style_spoilers.h
@@ -0,0 +1,5 @@
+#include "rpl/rpl.h"
+namespace Yukigram::Settings {
+extern rpl::variable<bool> *OldStyleSpoilers;
+extern const char kOptionOldStyleSpoilers[];
+}
diff --git a/Telegram/lib_ui/ui/effects/spoiler_mess.cpp b/Telegram/lib_ui/ui/effects/spoiler_mess.cpp
index d4413de547..5e834a11c4 100644
--- a/Telegram/lib_ui/ui/effects/spoiler_mess.cpp
+++ b/Telegram/lib_ui/ui/effects/spoiler_mess.cpp
@@ -10,6 +10,7 @@
 #include "ui/image/image_prepare.h"
 #include "ui/painter.h"
 #include "ui/integration.h"
+#include "styles/style_basic.h"
 #include "base/random.h"
 #include "base/flags.h"
 
@@ -428,14 +429,33 @@ SpoilerMessCached GenerateSpoilerMess(
 		size);
 }
 
+extern bool Yukigram_OldStyleSpoilers;
+bool Yukigram_OldStyleSpoilers = false;
+
+void FillSpoilerRectEx(QPainter &p, QRect rect, const SpoilerMessFrame &frame, QPoint originShift, bool forText = false);
+
 void FillSpoilerRect(
 		QPainter &p,
 		QRect rect,
 		const SpoilerMessFrame &frame,
 		QPoint originShift) {
+	return FillSpoilerRectEx(p, rect, frame, originShift);
+}
+
+void FillSpoilerRectEx(QPainter &p, QRect rect, const SpoilerMessFrame &frame, QPoint originShift, bool forText) {
 	if (rect.isEmpty()) {
 		return;
 	}
+	if (Yukigram_OldStyleSpoilers) {
+		if (!forText) {
+			p.fillRect(rect, Qt::transparent);
+			return;
+		}
+		auto path = QPainterPath();
+		path.addRoundedRect(rect, st::roundRadiusSmall, st::roundRadiusSmall);
+		p.fillPath(path, st::defaultTextPalette.spoilerFg->c);
+		return;
+	}
 	const auto &image = *frame.image;
 	const auto source = frame.source;
 	const auto ratio = style::DevicePixelRatio();
diff --git a/Telegram/lib_ui/ui/text/text_renderer.cpp b/Telegram/lib_ui/ui/text/text_renderer.cpp
index f89fc9e75d..2da7c5e314 100644
--- a/Telegram/lib_ui/ui/text/text_renderer.cpp
+++ b/Telegram/lib_ui/ui/text/text_renderer.cpp
@@ -18,6 +18,10 @@
 #include <private/qharfbuzz_p.h>
 #endif // Qt < 6.0.0
 
+namespace Ui {
+void FillSpoilerRectEx(QPainter &p, QRect rect, const SpoilerMessFrame &frame, QPoint originShift, bool forText = false);
+}
+
 namespace Ui::Text {
 namespace {
 
@@ -1619,7 +1623,7 @@ void Renderer::paintSpoilerRects(
 	const auto frame = _spoilerCache->lookup(color->c)->frame(index);
 	if (_spoilerCache) {
 		for (const auto &rect : rects) {
-			Ui::FillSpoilerRect(*_p, rect, frame, -rect.topLeft());
+			Ui::FillSpoilerRectEx(*_p, rect, frame, -rect.topLeft(), true);
 		}
 	} else {
 		// Show forgotten spoiler context part.
diff --git a/Telegram/lib_ui/ui/widgets/fields/custom_field_object.cpp b/Telegram/lib_ui/ui/widgets/fields/custom_field_object.cpp
index 67e2893d31..fa8d30e8cc 100644
--- a/Telegram/lib_ui/ui/widgets/fields/custom_field_object.cpp
+++ b/Telegram/lib_ui/ui/widgets/fields/custom_field_object.cpp
@@ -27,6 +27,8 @@ using SpoilerRect = InputFieldSpoilerRect;
 
 } // namespace
 
+void FillSpoilerRectEx(QPainter &p, QRect rect, const SpoilerMessFrame &frame, QPoint originShift, bool forText = false);
+
 class FieldSpoilerOverlay final : public RpWidget {
 public:
 	FieldSpoilerOverlay(
@@ -97,7 +99,7 @@ void FieldSpoilerOverlay::paintEvent(QPaintEvent *e) {
 		}
 		p->setOpacity(fgOpacity);
 		const auto shift = QPoint(0, *topShift) - rect.geometry.topLeft();
-		FillSpoilerRect(*p, rect.geometry, *frame, shift);
+		FillSpoilerRectEx(*p, rect.geometry, *frame, shift, true);
 	}
 }
 
