From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Thu, 28 May 2026 07:49:00 +0000
Subject: [PATCH] (Style) Fix button rounding in history top bar


diff --git a/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp b/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp
index 1e5bceb58e..2c9882db2c 100644
--- a/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_top_bar_widget.cpp
@@ -1159,6 +1159,7 @@ void TopBarWidget::updateControlsGeometry() {
 		};
 		auto first = (Ui::RoundButton*)(nullptr);
 		auto last = (Ui::RoundButton*)(nullptr);
+	if (false) {
 		for (const auto button : buttons) {
 			if (!button->isHidden()) {
 				if (!first) {
@@ -1167,6 +1168,7 @@ void TopBarWidget::updateControlsGeometry() {
 				last = button;
 			}
 		}
+	}
 		for (const auto button : buttons) {
 			if (button->isHidden()) {
 				continue;
