From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Thu, 28 May 2026 08:41:15 +0000
Subject: [PATCH] (Major) Retract quiz vote


diff --git a/Telegram/SourceFiles/history/view/history_view_context_menu.cpp b/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
index 0a029f8081..9cf3032015 100644
--- a/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
@@ -1782,7 +1782,6 @@ void FillPollOptionPage(
 		return;
 	}
 	if (!poll->closed()
-		&& !poll->quiz()
 		&& poll->voted()
 		&& !poll->revotingDisabled()) {
 		menu->addAction(
diff --git a/Telegram/SourceFiles/history/view/media/menu/history_view_poll_menu.cpp b/Telegram/SourceFiles/history/view/media/menu/history_view_poll_menu.cpp
index 3ea1053378..adabe7f516 100644
--- a/Telegram/SourceFiles/history/view/media/menu/history_view_poll_menu.cpp
+++ b/Telegram/SourceFiles/history/view/media/menu/history_view_poll_menu.cpp
@@ -197,8 +197,7 @@ void FillPollAnswerMenu(
 		}
 	}
 	if (!option.isEmpty()
-		&& !poll->closed()
-		&& !poll->quiz()) {
+		&& !poll->closed()) {
 		if (poll->voted()
 			&& !poll->revotingDisabled()) {
 			menu->addAction(
