From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Thu, 28 May 2026 08:41:15 +0000
Subject: [PATCH] (Major) Retract quiz vote


diff --git a/Telegram/SourceFiles/history/view/history_view_context_menu.cpp b/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
index ec6c2c24e2..68a88ca542 100644
--- a/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
+++ b/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
@@ -1814,7 +1814,6 @@ void FillPollOptionPage(
 		return;
 	}
 	if (!poll->closed()
-		&& !poll->quiz()
 		&& poll->voted()
 		&& !poll->revotingDisabled()) {
 		menu->addAction(
diff --git a/Telegram/SourceFiles/history/view/media/menu/history_view_poll_menu.cpp b/Telegram/SourceFiles/history/view/media/menu/history_view_poll_menu.cpp
index eb7f3757d4..2865ed8a60 100644
--- a/Telegram/SourceFiles/history/view/media/menu/history_view_poll_menu.cpp
+++ b/Telegram/SourceFiles/history/view/media/menu/history_view_poll_menu.cpp
@@ -198,8 +198,7 @@ void FillPollAnswerMenu(
 		}
 	}
 	if (!option.isEmpty()
-		&& !poll->closed()
-		&& !poll->quiz()) {
+		&& !poll->closed()) {
 		if (poll->voted()
 			&& !poll->revotingDisabled()) {
 			menu->addAction(
