From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Mon, 1 Jun 2026 15:07:23 +0000
Subject: [PATCH] (Shortcut) Normal send in context menu

---
 Telegram/SourceFiles/menu/menu_send.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Telegram/SourceFiles/menu/menu_send.cpp b/Telegram/SourceFiles/menu/menu_send.cpp
index 1e2bb9c495..f5f80464e8 100644
--- a/Telegram/SourceFiles/menu/menu_send.cpp
+++ b/Telegram/SourceFiles/menu/menu_send.cpp
@@ -741,6 +741,7 @@ FillMenuResult FillSendMenu(
 		? *iconsOverride
 		: st::defaultComposeIcons;
 
+	menu->addAction(tr::lng_send_button(tr::now), [=] { action({}, details); }, &st::menuIconSend);
 	if (sending && type != Type::Reminder) {
 		menu->addAction(
 			tr::lng_send_silent_message(tr::now),
-- 
2.53.0

