From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Wed, 3 Jun 2026 04:15:34 +0000
Subject: [PATCH] (Shortcut) Mention by name with right-click

Vanilla way (Ctrl + Left click) still works

diff --git a/Telegram/SourceFiles/chat_helpers/field_autocomplete.cpp b/Telegram/SourceFiles/chat_helpers/field_autocomplete.cpp
index 28b90efa6b..1834dce4c0 100644
--- a/Telegram/SourceFiles/chat_helpers/field_autocomplete.cpp
+++ b/Telegram/SourceFiles/chat_helpers/field_autocomplete.cpp
@@ -1574,6 +1574,8 @@ void FieldAutocomplete::Inner::mousePressEvent(QMouseEvent *e) {
 			_down = _sel;
 			_previewTimer.callOnce(QApplication::startDragTime());
 		}
+	} else if (e->button() == Qt::RightButton && !_mrows->empty() && !_overDelete) {
+		chooseSelected(FieldAutocomplete::ChooseMethod::ByClick);
 	}
 }
 
@@ -1901,7 +1903,7 @@ void InitFieldAutocomplete(
 		const auto user = data.user;
 		const auto ctrlClick = base::IsCtrlPressed()
 			&& data.method == FieldAutocomplete::ChooseMethod::ByClick;
-		if (data.mention.isEmpty() || ctrlClick) {
+		if (data.mention.isEmpty() || ctrlClick || QGuiApplication::mouseButtons() == Qt::RightButton) {
 			field->insertTag(
 				user->firstName.isEmpty() ? user->name() : user->firstName,
 				PrepareMentionTag(user));
