From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Wed, 3 Jun 2026 18:36:45 +0000
Subject: [PATCH] (Style) Constrain round messages width


diff --git a/Telegram/SourceFiles/history/view/media/history_view_gif.cpp b/Telegram/SourceFiles/history/view/media/history_view_gif.cpp
index 00529180dc..039041c51b 100644
--- a/Telegram/SourceFiles/history/view/media/history_view_gif.cpp
+++ b/Telegram/SourceFiles/history/view/media/history_view_gif.cpp
@@ -392,11 +392,17 @@ QSize Gif::countCurrentSize(int newWidth) {
 		auto reply = _parent->Get<Reply>();
 		auto forwarded = item->Get<HistoryMessageForwarded>();
 		if (via || reply || forwarded) {
+			const auto prevWidth = newWidth;
 			auto additional = additionalWidth(reply, via, forwarded);
 			newWidth += additional;
 			accumulate_min(newWidth, availableWidth);
 			auto usew = maxWidth() - additional;
 			auto availw = newWidth - usew - st::msgReplyPadding.left() - st::msgReplyPadding.left() - st::msgReplyPadding.left();
+			const auto diff = newWidth - prevWidth;
+			if (diff < 128) {
+				newHeight -= (128 - diff);
+				availw = 128;
+			}
 			if (!forwarded && via) {
 				via->resize(availw);
 			}
