From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Wed, 3 Jun 2026 18:36:45 +0000
Subject: [PATCH] (Style) Constrain round messages width


diff --git a/Telegram/SourceFiles/history/view/media/history_view_gif.cpp b/Telegram/SourceFiles/history/view/media/history_view_gif.cpp
index ebb84988a6..f515885973 100644
--- a/Telegram/SourceFiles/history/view/media/history_view_gif.cpp
+++ b/Telegram/SourceFiles/history/view/media/history_view_gif.cpp
@@ -428,6 +428,7 @@ QSize Gif::countCurrentSize(int newWidth) {
 		auto forwarded = item->Get<HistoryMessageForwarded>();
 		RefreshEphemeralPlate(_parent, _ephemeral.text);
 		if (via || reply || forwarded || !_ephemeral.text.isEmpty()) {
+			const auto prevWidth = newWidth;
 			auto additional = additionalWidth(reply, via, forwarded);
 			newWidth += additional;
 			accumulate_min(newWidth, availableWidth);
@@ -445,9 +446,14 @@ QSize Gif::countCurrentSize(int newWidth) {
 			const auto rectw = _ephemeral.onTop
 				? std::min(newWidth - st::msgReplyPadding.left(), additional)
 				: (newWidth - usew - st::msgReplyPadding.left());
-			const auto availw = rectw
+			auto availw = rectw
 				- st::msgReplyPadding.left()
 				- st::msgReplyPadding.left();
+			const auto diff = newWidth - prevWidth;
+			if (diff < 128) {
+				newHeight -= (128 - diff);
+				availw = 128;
+			}
 			if (!forwarded && via) {
 				via->resize(availw);
 			}
