From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <yuki.utk8g@gmail.com>
Date: Mon, 6 Apr 2026 02:29:13 +0300
Subject: [PATCH] (Style)(Opt) Custom emoji size

WIP

diff --git a/Telegram/SourceFiles/history/view/media/history_view_sticker.cpp b/Telegram/SourceFiles/history/view/media/history_view_sticker.cpp
index 67c31f4874..7075a61cfa 100644
--- a/Telegram/SourceFiles/history/view/media/history_view_sticker.cpp
+++ b/Telegram/SourceFiles/history/view/media/history_view_sticker.cpp
@@ -28,6 +28,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "core/core_settings.h"
 #include "core/click_handler_types.h"
 #include "window/window_session_controller.h"
+#include "yukigram/settings/custom_emoji_size.h"
 #include "data/data_session.h"
 #include "data/data_document.h"
 #include "data/data_document_media.h"
@@ -232,7 +233,7 @@ QSize Sticker::MessageEffectSize() {
 }
 
 QSize Sticker::EmojiSize() {
-	const auto side = std::min(st::maxAnimatedEmojiSize, kMaxEmojiSizeFixed);
+	const auto side = std::min(Yukigram::Settings::CustomEmojiSize->current(), kMaxEmojiSizeFixed);
 	return { side, side };
 }
 
diff --git a/Telegram/SourceFiles/yukigram/settings/custom_emoji_size.cpp b/Telegram/SourceFiles/yukigram/settings/custom_emoji_size.cpp
new file mode 100644
index 0000000000..1a4bb26f7b
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/custom_emoji_size.cpp
@@ -0,0 +1,17 @@
+#include "yukigram/settings/custom_emoji_size.h"
+#include "yukigram/options.h"
+
+namespace Yukigram::Settings {
+
+rpl::variable<int> *CustomEmojiSize = Yukigram::Options::make<int>(kOptionCustomEmojiSize, {
+	.keywords = { u"yukigram"_q, u"emoji"_q, u"size"_q },
+	.category = "style",
+	.restartRequired = true,
+	.defaultValue = 112,
+	.checkValid = Yukigram::Options::Valid::Bounded(32, 256),
+	.validHint = Yukigram::Options::Valid::BoundedHint(32, 256),
+});
+
+const char kOptionCustomEmojiSize[] = "custom-emoji-size";
+
+}
diff --git a/Telegram/SourceFiles/yukigram/settings/custom_emoji_size.h b/Telegram/SourceFiles/yukigram/settings/custom_emoji_size.h
new file mode 100644
index 0000000000..a39fe340ab
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/custom_emoji_size.h
@@ -0,0 +1,5 @@
+#include "rpl/rpl.h"
+namespace Yukigram::Settings {
+extern rpl::variable<int> *CustomEmojiSize;
+extern const char kOptionCustomEmojiSize[];
+}
