From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Thu, 4 Jun 2026 03:00:06 +0000
Subject: [PATCH] (Style)(Opt) Round message size


diff --git a/Telegram/SourceFiles/history/view/media/history_view_gif.cpp b/Telegram/SourceFiles/history/view/media/history_view_gif.cpp
index 286f01a1ed..67f4540417 100644
--- a/Telegram/SourceFiles/history/view/media/history_view_gif.cpp
+++ b/Telegram/SourceFiles/history/view/media/history_view_gif.cpp
@@ -51,6 +51,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "ui/ui_utility.h"
 #include "ui/effects/path_shift_gradient.h"
 #include "ui/effects/spoiler_mess.h"
+#include "yukigram/settings/round_size.h"
 #include "data/data_photo.h"
 #include "data/data_photo_media.h"
 #include "data/data_session.h"
@@ -313,7 +314,7 @@ QSize Gif::countThumbSize(int &inOutWidthMax) const {
 		} else if (_data->isVideoFile()) {
 			return st::maxMediaSize;
 		} else if (_data->isVideoMessage()) {
-			return st::maxVideoMessageSize;
+			return Yukigram::Settings::RoundSize->current();
 		}
 		return st::maxGifSize;
 	}();
diff --git a/Telegram/SourceFiles/yukigram/settings/round_size.cpp b/Telegram/SourceFiles/yukigram/settings/round_size.cpp
new file mode 100644
index 0000000000..0ed04e4459
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/round_size.cpp
@@ -0,0 +1,17 @@
+#include "yukigram/settings/round_size.h"
+#include "yukigram/options.h"
+
+namespace Yukigram::Settings {
+
+rpl::variable<int> *RoundSize = Yukigram::Options::make<int>(kOptionRoundSize, {
+	.keywords = { u"yukigram"_q, u"round"_q, u"video"_q, u"size"_q },
+	.category = "style",
+	.restartRequired = true,
+	.defaultValue = 240,
+	.checkValid = Yukigram::Options::Valid::Bounded(80, 384),
+	.validHint = Yukigram::Options::Valid::BoundedHint(80, 384),
+});
+
+const char kOptionRoundSize[] = "round-size";
+
+}
diff --git a/Telegram/SourceFiles/yukigram/settings/round_size.h b/Telegram/SourceFiles/yukigram/settings/round_size.h
new file mode 100644
index 0000000000..dc5aa4aae5
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/round_size.h
@@ -0,0 +1,5 @@
+#include "rpl/rpl.h"
+namespace Yukigram::Settings {
+extern rpl::variable<int> *RoundSize;
+extern const char kOptionRoundSize[];
+}
