From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Fri, 3 Jul 2026 21:44:41 +0000
Subject: [PATCH] (Chat)(Opt) Disable swipe-next-channel


diff --git a/Telegram/SourceFiles/history/history_view_pull_to_next_channel.cpp b/Telegram/SourceFiles/history/history_view_pull_to_next_channel.cpp
index f9d330c623..cfd54d0295 100644
--- a/Telegram/SourceFiles/history/history_view_pull_to_next_channel.cpp
+++ b/Telegram/SourceFiles/history/history_view_pull_to_next_channel.cpp
@@ -25,6 +25,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "main/main_session.h"
 #include "storage/storage_shared_media.h"
 #include "support/support_preload.h"
+#include "yukigram/settings/disable_swipe_next.h"
 #include "ui/chat/chat_style.h"
 #include "ui/chat/continuous_scroll.h"
 #include "ui/widgets/elastic_scroll.h"
@@ -597,6 +598,9 @@ void PullToNextChannel::setHistory(History *history) {
 }
 
 bool PullToNextChannel::active() const {
+	if (Yukigram::Settings::DisableSwipeNext->current()) {
+		return false;
+	}
 	return _history
 		&& _history->peer->isBroadcast()
 		&& atBottom()
diff --git a/Telegram/SourceFiles/yukigram/settings/disable_swipe_next.cpp b/Telegram/SourceFiles/yukigram/settings/disable_swipe_next.cpp
new file mode 100644
index 0000000000..3cae8ec276
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/disable_swipe_next.cpp
@@ -0,0 +1,13 @@
+#include "yukigram/settings/disable_swipe_next.h"
+#include "yukigram/options.h"
+
+namespace Yukigram::Settings {
+
+rpl::variable<bool> *DisableSwipeNext = Yukigram::Options::make<bool>(kOptionDisableSwipeNext, {
+	.keywords = { u"yukigram"_q, u"swipe"_q, u"next"_q, u"channel"_q, u"unread"_q, u"bottom"_q },
+	.category = "chat",
+});
+
+const char kOptionDisableSwipeNext[] = "disable-swipe-next";
+
+}
diff --git a/Telegram/SourceFiles/yukigram/settings/disable_swipe_next.h b/Telegram/SourceFiles/yukigram/settings/disable_swipe_next.h
new file mode 100644
index 0000000000..2eb31b1219
--- /dev/null
+++ b/Telegram/SourceFiles/yukigram/settings/disable_swipe_next.h
@@ -0,0 +1,5 @@
+#include "rpl/rpl.h"
+namespace Yukigram::Settings {
+extern rpl::variable<bool> *DisableSwipeNext;
+extern const char kOptionDisableSwipeNext[];
+}
