From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zoh-li <i@zoh.li>
Date: Sun, 21 Jun 2026 15:44:21 +0330
Subject: [PATCH] (Interface) Show custom title in group call bar


diff --git a/Telegram/SourceFiles/ui/chat/group_call_bar.cpp b/Telegram/SourceFiles/ui/chat/group_call_bar.cpp
index 13481b99dd..db62cde0b8 100644
--- a/Telegram/SourceFiles/ui/chat/group_call_bar.cpp
+++ b/Telegram/SourceFiles/ui/chat/group_call_bar.cpp
@@ -293,17 +293,17 @@ void GroupCallBar::paintTitleAndStatus(Painter &p) {
 		left,
 		titleTop,
 		width,
-		(!_content.scheduleDate
+		(!_content.title.isEmpty()
+			? ((titleWidth > available)
+				? font->elided(_content.title, available)
+				: _content.title)
+			: !_content.scheduleDate
 			? (_content.livestream
 				? tr::lng_group_call_title_channel
 				: tr::lng_group_call_title)(tr::now)
-			: _content.title.isEmpty()
-			? (_content.livestream
+			: (_content.livestream
 				? tr::lng_group_call_scheduled_title_channel
-				: tr::lng_group_call_scheduled_title)(tr::now)
-			: (titleWidth > available)
-			? font->elided(_content.title, available)
-			: _content.title));
+				: tr::lng_group_call_scheduled_title)(tr::now)));
 	p.setPen(st::historyStatusFg);
 	p.setFont(st::defaultMessageBar.text.font);
 	const auto when = [&] {
